app/usecase/user.go

```diff
@@ -32,3 +32,25 @@ func (u User) Signup(user entities.User) (string, error) {
 
 	return id, nil
 }
+
+// Delete represents business logic
+// and will take care of deleting user.
+func (u User) Delete(user entities.UserID) error {
```

Comment:
In our case, why not just?
```go
func (u User) Delete(user entities.UserID) error {
	return u.Repo.Delete(user)
}
```

Reply:
Your version is really better. Thanks, fixed